index.bs

```diff
@@ -1118,6 +1118,76 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "S
+:: In order to implement the algorithms defined in [[#sctn-rp-operations]],
+    the [=[RP]=] MUST store some properties of registered [=public key credential sources=].
+    The [=credential record=] [=struct=] is an abstraction of these properties stored in a [=user account=].
+    A&nbsp;credential record is created during a [=registration ceremony=] and used in subsequent [=authentication ceremonies=].
+    [=[RPS]=] MAY delete credential records as necessary or when requested by users.
+
+    The following [=struct/items=] are RECOMMENDED in order to implement all steps of
+    [[#sctn-registering-a-new-credential]] and [[#sctn-verifying-assertion]] as defined:
+
+    <dl dfn-for="credential record">
+        :   <dfn>type</dfn>
+        ::  The [=public key credential source/type=] of the [=public key credential source=].
+
+        :   <dfn>id</dfn>
+        ::  The [=Credential ID=] of the [=public key credential source=].
+
+        :   <dfn>publicKey</dfn>
+        ::  The [=credential public key=] of the [=public key credential source=].
+
+        :   <dfn>signCount</dfn>
+        ::  The latest value of the [=authData/signCount|signature counter=] in the [=authenticator data=]
+            from any [=ceremony=] using the [=public key credential source=].
+
+        :   <dfn>transports</dfn>
+        ::  The value returned from {{AuthenticatorAttestationResponse/getTransports()}}
+            when the [=public key credential source=] was [=registration|registered=].
+
+            Note: Modifying or removing [=list/items=] from the value returned from {{AuthenticatorAttestationResponse/getTransports()}}
+            could negatively impact user experience, or even prevent use of the corresponding credential.
+
+        :   <dfn>BE</dfn>
+        ::  The value of the [=authData/flags/BE=] [=flag=] when the [=public key credential source=] was created.
+
+        :   <dfn>BS</dfn>
+        ::  The latest value of the [=authData/flags/BS=] [=flag=] in the [=authenticator data=]
+            from any [=ceremony=] using the [=public key credential source=].
+    </dl>
+
+    The following [=struct/items=] are OPTIONAL:
+
+    <dl dfn-for="credential record">
+        :   <dfn>attestationObject</dfn>
+        ::  The value of the {{AuthenticatorAttestationResponse/attestationObject}} attribute
+            when the [=public key credential source=] was [=registration|registered=].
+            Storing this enables the [=[RP]=] to reference the credential's [=attestation statement=] at a later time.
+
+        :   <dfn>attestationClientDataJSON</dfn>
+        ::  The value of the {{AuthenticatorResponse/clientDataJSON}} attribute
+            when the [=public key credential source=] was [=registration|registered=].
+            Storing this in combination with the above [=credential record/attestationObject=] [=struct/item=]
+            enables the [=[RP]=] to re-verify the [=attestation signature=] at a later time.
+    </dl>
```

Comment:
@emlun now that attestation is possible on getAssertion, do we need additional language here to address that?

Reply:
Yes, but I'd like to tie that into #1663 instead since that's where that change will be introduced. I intend to open a meta-PR to update #1663 to use the new credential record abstraction once this is merged (unless #1663 gets merged before I can get to it :smile:).